### PR TITLE
add missing -lrt flag to Makefile

### PR DESCRIPTION
**Pull request by miguelraz**
vis #140

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -66,7 +66,7 @@ else ifneq ($(OS), Darwin)
   else
     MIMALLOC_LIB = out/mimalloc/libmimalloc.a
     CPPFLAGS += -Ithird-party/mimalloc/include
-    LIBS += -Wl,-whole-archive $(MIMALLOC_LIB) -Wl,-no-whole-archive
+    LIBS += -Wl,-whole-archive $(MIMALLOC_LIB) -Wl,-no-whole-archive -lrt
```
Comment (rui314) [inline Makefile:69]: Don't you want to move this to line 29 instead of this line?

**Review by rui314 — CHANGES_REQUESTED** [review]
Please read CONTRIBUTING.md and add a signed-off-by line.
Thanks!

File: Makefile (hunk 1 above)
Comment (miguelraz) [reply Makefile]: Fair, wasn't sure where precisely. Thanks!

**Comment by miguelraz** [comment]
Newest commit is done with `git commit -s`, feel free to squash and merge.

**Review by thesamesam — APPROVED** [review]
lgtm

**Comment by rui314** [comment]
It looks like it breaks macOS build, but I'll fix that after merging this PR. Thank you for doing this!

**Comment by thesamesam** [comment]
@rui314 As time goes on, I anticipate having to add/remove more of these hacks based on platform (e.g. explicit/implicit pthread and so on). How do you feel about me looking at meson? no promises, but just had an idea I've had the last few days. It'd handle stuff like this for us.

**Comment by rui314** [comment]
@thesamesam At this moment, I'd like to stick with the good ol' makefile. Everybody knows how to use the make command, and the current state of our makefile isn't messy IMO. In the future, especially when porting it to non-Unix platform (i.e. Windows), we'll probably have to start using a more sophisticated, cross-platform build system such as cmake. But for now, I think makefile works fine for us.

**Comment by thesamesam** [comment]
@rui314 No problem. I agree it's not really messy right now. Let me know if/when that changes. We'll see what happens if any more bug reports come in and such too. Thank you!

**Comment by miguelraz** [comment]
Thank you for working on this!
I'll wanna ship this a recipe for this with BinaryBuilder.jl so that Julia users don't have to worry about installing it.

**Pull request merged**